front/src/modules/simulationResult/components/ManchetteWithSpaceTimeChart/ManchetteWithSpaceTimeChart.tsx

```diff
@@ -45,9 +51,94 @@ const ManchetteWithSpaceTimeChartWrapper = ({
 
   const [waypointsPanelIsOpen, setWaypointsPanelIsOpen] = useState(false);
 
+  // Cut the space time chart curves if the first or last waypoints are hidden
+  const spaceTimeChartLayersData = useMemo(() => {
+    let filteredProjectPathTrainResult = projectPathTrainResult;
+    let filteredConflicts = conflicts;
+
+    if (!waypointsPanelData) return { filteredProjectPathTrainResult, filteredConflicts };
+
+    const { filteredWaypoints } = waypointsPanelData;
+    const firstPositon = filteredWaypoints.at(0)!.position;
+    const lastPosition = filteredWaypoints.at(-1)!.position;
+
+    if (
+      filteredWaypoints.length >= 2 &&
+      (firstPositon !== 0 || lastPosition !== operationalPoints.at(-1)!.position)
+    ) {
+      filteredProjectPathTrainResult = projectPathTrainResult.map((train) => ({
+        ...train,
+        space_time_curves: train.space_time_curves.map(({ positions, times }) => {
+          const cutPositions = new Set<number>();
+          const cutTimes = new Set<number>();
```

Comment:
Why use a `Set` here instead of an `Array`?

A `Set` will forbid multiple positions/times with the same value. If a curve is not strictly monotonic (e.g. if a curve goes down and then up, or is constant), then we'd drop some data points. Also I think there is an assumption that there are as many position values are there are time values.

Reply:
You're right done